MP-15 Integration with React Hook Form

kacpergadula opened the pull request.

kacpergadula committed:
MP-15 Integration with React Hook Form

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
+const AddSupplierData = () => {
+  const { register, handleSubmit, errors } = useForm();
+
+  const StylesS = styled.span`
```
If I'm not mistaken, @dominikmlynarczyk  talked about naming such elements as `StyledSpan` meaning Styled + tag / component name.

kacpergadula replied:
I will do it this way. It will be more legible

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
+                  ref={register({ required: 'Brak nazwy' })}
+                />
+                {errors.name && (
+                  <StylesS>
```
`StylesS` is declared as (styled) span already so you don't need inner duplication.

mfurmanit replied:
That should be corrected all the way down this file.

kacpergadula replied:
I will correct this

mfurmanit commented on tsconfig.json:
```diff
@@ -19,7 +19,8 @@
     "jsx": "react-jsx",
-    "baseUrl": "src"
+    "baseUrl": "src",
+    "noImplicitAny": false
```
Use proper type-definitions and TypeScript capabilities. We shouldn't use implicit any unless it would be absolutely required.

kacpergadula replied:
sure, I'll delete it

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
+              </Container>
+              <NumberInput>
+                <NumberInputField
+                  name="telefonNumber"
```
Should be `telephoneNumber`. Let's not confuse two languages.

kacpergadula replied:
I will correct this

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
+                  </Container>
+                  <Box>
+                    <Input
+                      name="buildNumber"
```
Should be `buildingNumber`.

kacpergadula replied:
I will correct this

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
+                      border="1px"
+                      borderRadius="2em"
+                      borderColor="gray.300"
+                      ref={register({ minLength: 1, required: 'Nie podano miasta' })}
```
ulicy*
This block should be formatted exactly like the others higher.

kacpergadula replied:
I will correct this

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
+                  <Box>
+                    <Input
+                      w="16em"
+                      name="kodNumber"
```
Should be just `postalCode`.

kacpergadula replied:
I will correct this

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
+    padding: 0.5rem 0 0 0;
+  `;
+
+  const onSubmit = (data) => {
```
`data` should be typed, even with temporal interface / type.

kacpergadula replied:
You are right. I forgot to declare the type `data`

mfurmanit replied:
@kacpergadula I meant creating an interface that mirrors the data on the form, but for now it can stay that way.

mfurmanit commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,213 @@
-    </Box>
-  </Flex>
-);
+const AddSupplierData = () => {
```
@kacpergadula Try to make this component responsive. I saw that it is not right now.

kacpergadula replied:
I corrected this

mfurmanit reviewed (changes requested):
Good job, please address all the issues, resolve conflicts or write comments.

kacpergadula committed:
MP-15 Correction of: Input names,Styled Components

kacpergadula committed:
MP-15 form responsive components

kacpergadula committed:
chore: :bug: repairing file conflicts

repairing file conflicts ( package-lock.json , package.json , src/AddSupplier.tsx )

repairing file conflicts

dominikmlynarczyk commented on package.json:
```diff
@@ -12,16 +12,16 @@
     "framer-motion": "^3.10.4",
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
+    "react-hook-form": "^6.15.5",
```
react-hook-form [released major v7 upgrade version three weeks ago](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.0.0) and it has important [breaking changes](https://github.com/react-hook-form/react-hook-form/discussions/3714) (such as `ref` prop being deprecated). Also it takes much ephasis on strict type checking, so I strongly recommend upgrade now, cause we will use this library extensively all over mfg-pos.

@kacpergadula - check migration guide [here](https://react-hook-form.com/migrate-v6-to-v7/)

dominikmlynarczyk commented on src/components/Supplier/AddSupplier.tsx:
```diff
@@ -1,22 +1,34 @@
-  </Flex>
-);
+const Form = () => {
+  const StyledDiv = styled.div`
```
I'm considering dropping Styled Components altogether. I've familiarized myself a bit with Chakra and it uses [Emotion](https://emotion.sh/docs/introduction) under the hood, it's a different "CSS in JS" library, so it probably doesn't make much sense to use Styled Components and Emotion together.

Referring directly to the code - why are you writing a special Styled Component just to set the width and padding? Note that below you are using components from Chakra and applying the props that Chakra provides to achieve the same effect. 

The purpose of going into a UI library is to follow conventions, design system, etc., so creating this type of Styled Component makes absolutely no sense. As for the responsiveness issue, please [read the Chakra documentation](https://chakra-ui.com/docs/features/responsive-styles), [here you can find examples](https://chakra-ui.com/docs/features/responsive-styles#more-examples) of how to apply responsiveness.

dominikmlynarczyk commented on src/components/Supplier/AddSupplierButtons.tsx:
```diff
@@ -1,14 +1,29 @@
+import styled from 'styled-components';
+
+const AddSupplierButtons = () => {
+  const StyledBox = styled(Box)`
```
Same situation as above, consider using native Chakra props instead.

dominikmlynarczyk commented on src/components/Supplier/AppSupplierData/AddSupplierData.tsx:
```diff
@@ -1,17 +1,203 @@
+                  borderRadius="2em"
+                  borderColor="gray.300"
+                  name="name"
+                  ref={register({ required: 'Brak nazwy' })}
```
Whenever you are using such things as `ref` prop, from `useForm()` hook, upgrade it to v7 API.
Do the same with other deprecated parts of react-hook-form API as I mentioned above.

dominikmlynarczyk commented on src/styled/Supplier/AppSupplierData/StyledGrid.tsx:
```diff
@@ -0,0 +1,12 @@
+import { Grid } from '@chakra-ui/react';
+import styled from 'styled-components';
+
+const StyledGrid = styled(Grid)`
```
Same situation as above, consider using native Chakra props instead.

dominikmlynarczyk commented on src/styled/Supplier/AppSupplierData/StyledSpan.tsx:
```diff
@@ -0,0 +1,29 @@
+import styled from 'styled-components';
+
+const StyledSpan = styled.span`
```
Same situation as above, consider using native Chakra props instead.
Please get familar with Chakra [theming](https://chakra-ui.com/docs/theming/theme), in this case Chakra is providing [typography solution](https://chakra-ui.com/docs/theming/theme#typography) for you.

dominikmlynarczyk commented on src/styled/Supplier/AppSupplierData/StyledVStack.tsx:
```diff
@@ -0,0 +1,10 @@
+import { Grid } from '@chakra-ui/react';
+import styled from 'styled-components';
+
+const StyledVStack = styled(Grid)`
```
Same situation as above. Read RWD approach at Chakra docs.

dominikmlynarczyk commented on src/styled/Supplier/AppSupplierData/StyledWrapItem.tsx:
```diff
@@ -0,0 +1,13 @@
+import { WrapItem } from '@chakra-ui/react';
+import styled from 'styled-components';
+
+const StyledWrapItem = styled(WrapItem)`
```
Same situation as above, consider using native Chakra props instead.

dominikmlynarczyk reviewed (changes requested):
Thank you for your PR! 
Read all the information and apply it in your code.

kacpergadula committed:
feat: :arrow_up: migration react-hook-form from v6 to v7

migration react-hook-form from v6 to v7 : implementation of new errors and register, definition of useForm types

kacpergadula committed:
style: :art: chakra responsive styles

implementation of chakra responsive styles

dominikmlynarczyk commented:
Didn't investigate much on the migration part, but all ref occurrences changed to register look good.
I don't review style changes as we will apply styling from atomic components.
